### PR TITLE
Add once, negate, memoize.

### PR DESCRIPTION
**Pull request by Wojbie**

### CODE REVIEW

File: doc/negate.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+* _.negate(f)
```
Comment (SquidDev) [inline doc/negate.txt:1]: While negation does apply to boolean algebra, one does risk confusing it with unary minus. I'd say complement, but that sounds too much like sets. Maybe `inverted`?

File: doc/once.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+* _.once(f)
```
Comment (SquidDev) [inline doc/once.txt:1]: Why `once` rather than `lazy`? Allowing arguments seems a wee bit confusing too, but eh.

**Review by SquidDev — CHANGES_REQUESTED** [review]

File: doc/once.txt (hunk 2 above)
Comment (Lemmmy) [reply doc/once.txt:1]: same names as lodash

File: doc/negate.txt (hunk 1 above)
Comment (Lemmmy) [reply doc/negate.txt:1]: https://lodash.com/docs/4.17.10#negate

File: doc/once.txt (hunk 2 above)
Comment (Lemmmy) [reply doc/once.txt:1]: https://lodash.com/docs/4.17.10#once

Comment (plt-amy) [reply doc/once.txt:1]: I agree with squid, `lazy` seems like a better name. We aren't lodash

File: library.lua [hunk 3]
```diff
@@ -255,6 +255,36 @@ function _.values(t)
...
+        end
+        return unpack(cache,1,cache.n)
+    end
+end
+
+function _.memoize(f)
+    _.expect('partial', 1, 'function', f)
+    local cache = {}
+    return function(...)
+        local key = select(1, ... )
```
Comment (plt-amy) [inline library.lua:273]: Why does memoize ignore the rest of the arguments? No good.

Comment (Wojbie) [reply library.lua:273]: As stated before i was following the lodash docs. I guess making it memorize all args would be workable if you want it that way?

Comment (plt-amy) [reply library.lua:273]: We're not lodash. Yes, do that.

**Pull request closed without merging**